salt/addons/dex/pre.sls

```diff
@@ -0,0 +1,27 @@
+{% from '_macros/certs.jinja' import alt_master_names, certs with context %}
+
+{% set dex_alt_names = ["dex",
+                        "dex.kube-system",
+                        "dex.kube-system.svc",
+                        "dex.kube-system.svc." + pillar['internal_infra_domain']] %}
+{{ certs('dex',
+         pillar['ssl']['dex_crt'],
+         pillar['ssl']['dex_key'],
+         cn = 'Dex',
+         extra_alt_names = alt_master_names(dex_alt_names)) }}
+
+# This file needs to be explicitly copied over *before* ./main.sls runs
+# kubectl_apply_dir_template(). Otherwise one of the templates attempts to
+# access the file, and it does not exist yet, breaking the orchestration.
+# Unfortunately using the default include mechanism of salt, to force order
+# doesn't seem to work when running via orchestrate runner, so this is
+# split in different files that the orch runner calls with explicit
+# dependencies.
```

Comment:
Isn't enough with an `include` **and** a `require` in the state that requires it? I find this split in different files a bit confusing...